### PR TITLE
Remove unneeded mapviews reinstantiations (remote dataviews performance)

### PR DESCRIPTION
**Pull request by padawannn**
- Remove TILES_LOADED event (use DATA_CHANGED instead). I keep this event commented in case we use it in the future.
- I've tried different approaches and with the current code, the easier way is to have a cache in MapsDataviews.ts. Another approach requires a big refactor

### CODE REVIEW

File: src/maps/MapsDataviews.ts [hunk 1]
```diff
@@ -55,18 +57,27 @@ export class MapsDataviews {
...
     dataviewName: string,
     type: string,
     options: Partial<MapDataviewsOptions>
   ) {
+    const layergroupCacheKey = `type_${JSON.stringify(options)}`;
+
+    if (this._layergroupCache[layergroupCacheKey]) {
+      return this._layergroupCache[layergroupCacheKey];
+    }
+
```
Comment (VictorVelarde) [inline src/maps/MapsDataviews.ts:80]: I see the trick, nice cache

File: src/viz/layer/Layer.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ import { basicStyle } from '../style/helpers/basic-style';
 export enum LayerEvent {
   DATA_READY = 'dataReady',
   DATA_CHANGED = 'dataChanged',
-  TILES_LOADED = 'tilesLoaded',
+  // TILES_LOADED = 'tilesLoaded',
```
Comment (VictorVelarde) [inline src/viz/layer/Layer.ts]: So eventually no use at all of this...? 

I guess that if we need to 'refresh' the tiles without changing the viewport (for example when requiring new fields in the map), we are rising an equivalent data changed event right?

If that's it, we should completely remove TILES_LOADED event

**Review by VictorVelarde — APPROVED** [review]
LGTM. I would remove TILES_LOADED at all, if we check it's not required anymore.

You can push this one first, and then I will try to apply the maps refactor from https://github.com/CartoDB/web-sdk/pull/137

File: src/viz/layer/Layer.ts (hunk 2 above)
Comment (padawannn) [reply src/viz/layer/Layer.ts]: ok, I've removed the comments

**Pull request merged**